src/components/Header/Header.tsx

```diff
@@ -0,0 +1,25 @@
+import { Flex } from "@adobe/react-spectrum";
+import { ToggleButton } from "@adobe/react-spectrum";
+import ClassicGridView from "@spectrum-icons/workflow/ClassicGridView";
+import React, { FC } from "react";
+
+interface Props {
+    isToggled?: boolean;
+}
+
+const Header: FC<Props> = ({ isToggled }) => {
+    const [selected, setSelected] = React.useState(isToggled);
```

Comment:
En casos como este donde `isToggled` es opcional _**pero**_ se usa para inicializar un valor, es mejor ponerle un default, fallback value 
```suggestion
interface Props {
    isToggled?: boolean;
}

const Header: FC<Props> = ({ isToggled = false }) => {
    const [selected, setSelected] = React.useState(isToggled);
```